core/runtime.rs

```diff
@@ -728,10 +728,13 @@ impl JsRuntime {
   /// Initializes JS of provided Extensions in the given realm
   fn init_extension_js(&mut self, realm: &JsRealm) -> Result<(), Error> {
     // Take extensions to avoid double-borrow
-    let mut extensions: Vec<Extension> =
-      std::mem::take(&mut self.extensions_with_js);
-    for m in extensions.iter_mut() {
-      let js_files = m.init_js();
+    let extensions = std::mem::take(&mut self.extensions_with_js);
+    for (ext, previous_exts) in extensions
+      .iter()
+      .enumerate()
+      .map(|(i, ext)| (ext, &extensions[..i]))
+    {
+      let js_files = ext.init_js(previous_exts);
```

Comment:
I think this validation should happen in `JsRuntime::new()` before we call `init_extension_ops()` - ideally in `JsRuntime::collect_ops()`.